### ISSUE TITLE
Occasional computer freeze

### ISSUE THREAD

**Issue by igor2x** [mlutfy/hidetopbar#266]
Hi,
I installed HideTopBar few months ago. Excellent extension, that hides Gnome top bar that I see as a clutter that wastes my precious small laptop screen. I can always get top bar with pressing Windows key.

The problem I have is my laptop with Ubuntu 20.04 (that is Gnome Shell 3.36.4 with X11) freezes time to time, sometimes two, three times a day, then everything works fine for 20 days and then freeze reappears. I can't reproduce the problem, it just happens out of the blue.

I have reported problem in Ubuntu bug tracker: https://bugs.launchpad.net/ubuntu/+source/gnome-shell/+bug/1903986 and the response is, he suspects from system logs files that most likely cause of my laptop freezes is HideTopBar extension.

Today laptop freeze appeared when in "Keyboard shortcut" section of in HideTopBar extension I tried to change "Key that triggers the bar to be shown." But in my case most of the time I have laptop freezes in Remmina remote desktop tool using it in full-screen (because I am working from home).

Why freeze HideTopBar is a suspect? Because at the bottom of above bug report (link above) in 34 using command:
`journalctl -b-1 `
and error logged was:

> jan 04 10:03:20 tp gjs[141945]: JS ERROR: TypeError: model.get is not a function
>                                 buildPrefsWidget/<@/home/igor/.local/share/gnome-shell/extensions/hidetopbar@mathieu.bidon.ca/prefs.js:229:31
>                                 _init/createCheckedMethod/<@resource:///org/gnome/gjs/modules/core/overrides/Gio.js:529:46
>                                 buildPrefsWidget/<@/home/igor/.local/share/gnome-shell/extensions/hidetopbar@mathieu.bidon.ca/prefs.js:200:18
>                                 run@resource:///org/gnome/Shell/Extensions/js/dbusService.js:175:20
>                                 main@resource:///org/gnome/Shell/Extensions/js/main.js:19:13
>                                 run@resource:///org/gnome/gjs/modules/script/package.js:222:19
>                                 start@resource:///org/gnome/gjs/modules/script/package.js:206:5
>                                 @/usr/share/gnome-shell/org.gnome.Shell.Extensions:1:17

Then my laptop has frozen and I had to hard press laptops power-off button, to reboot system.

Current HideTopBar extension version that is installed is 81, which from extension web page see is the latest one.
Regards

**Comment by tuxor1337** [comment]
I don't think that this error message is to blame and I doubt that Hide Top Bar is causing your problems. But I still pushed a fix for the TypeError that you quoted.

**Comment by igor2x** [comment]
Thanks for fixing this. I have installed HideTopBar from source from https://github.com/mlutfy/hidetopbar
Interesting at last step: `gnome-shell --replace &` Gnome Shell crashed with "Ops..." crash error. I have rebooted Ubuntu and now it looks fine.

I have turned HideTopBar on. I can't live without it. Gnome's top bar is really annoying to me.

**Comment by jacksongoode** [comment]
I've also been attempting to track down what will cause an application to freeze. So far I've noticed this in Spotify and Chromium (ungoogled) it seems to unfreeze when I press Super to open the launcher (I'm using Arc so I'm wondering maybe if it has to do with #296).

**Comment by igor2x** [comment]
Sorry for late reply. My computer freeze problem is not related to HideTopBar extension. In my case there is some problem with x.org. I switched Ubuntu to Wayland few months and crashes stopped immediately. After few months I have turned x.org back again and crashed appear again. Switching back to Wayland and now I don't have crashes any more.

I am closing this bug report down, because it was not related to HideTopBar extension.
